pkg/provider/provider_test.go

```diff
@@ -49,11 +51,13 @@ func TestProvider_SetData(t *testing.T) {
 		key := []byte{1, 1, 1}
 		value := []byte{0, 1, 3}
 		encodedValue := []byte{0, 1, 3, 5, 34}
+		encodedKey := []byte{0, 1, 3, 5, 34}
```

Comment:
Usually try to use different values for test variables because if you do not then there is a possibility that your tests won't catch an error when you mistyped/switched parameters inside the implementation.